tests/testthat/test_end_to_end.R

```diff
@@ -0,0 +1,55 @@
+test_data <- structure(list(x = 1:21, 
```

Comment:
@ImogenConnorHelleur I think the test data should be of the correct data type for the charts being tested - i.e. a c-chart only makes sense for integer counts (since the Poisson distribution is only well defined for integers), a p-chart only makes sense for integer numerator and denominator (since the binomial distribution is only well defined in this case). Could you replace the test data to address this, in both this test file and the others testing the SPC functions?